src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -81,6 +81,8 @@ object FunctionRegistry {
             None
         }
 
+        rtTypConcrete.foreach(f.retType.unify(_))
```

Comment:
This ignores the return type of `unify`, in particular, if the unification fails (i.e. a method only works for `Variant(GRCh38)` and `rtTypConcrete` is a `GRCh37` variant) the lookup must exclude this `f`.